float32 fDeltaEncoder1
float32 fDeltaEncoder2
